jobs/mongodb_migration/src/mongodb_migration/migrations/_20230216112500_cache_split_names_from_streaming.py

```diff
@@ -1,44 +1,23 @@
 # SPDX-License-Identifier: Apache-2.0
-# Copyright 2022 The HuggingFace Authors.
+# Copyright 2023 The HuggingFace Authors.
 
 import logging
 
 from libcommon.constants import CACHE_COLLECTION_RESPONSES, CACHE_MONGOENGINE_ALIAS
-from libcommon.simple_cache import CachedResponse
 from mongoengine.connection import get_db
 
-from mongodb_migration.check import check_documents
-from mongodb_migration.migration import Migration
-
-split_names = "/split-names"
-split_names_from_streaming = "/split-names-from-streaming"
-split_names_tmp = "/split-names-TMP"
+from mongodb_migration.renaming_migrations import CacheRenamingMigration
 
 
 # connection already occurred in the main.py (caveat: we use globals)
-class MigrationCacheUpdateSplitNames(Migration):
+class MigrationCacheUpdateSplitNames(CacheRenamingMigration):
     def up(self) -> None:
-        logging.info(f"Rename cache_kind field from {split_names} to {split_names_from_streaming}")
+        tmp_kind = f"{self.cache_kind}-TMP"
```

Comment:
is the intermediate step actually required now? or this migration can also be refactored as simple `CacheRenamingMigration`?

Reply:
I don't know. The idea was to ensure that, in the case some cache entries already existed with the new kind, they would not be overwritten with older entries. But since the migration should be blocking the upgrade, thus: the new cache kind should not had been used already, I don't remember why we added this. @AndreaFrancis do you remind?

Reply:
I can just add this step to the common `CacheRenamingMigration` class just in case, but if @AndreaFrancis could explain what the case is exactly :D and if it's possible to have collisions like this, would be good for me to understand.
And actually, aren't new kind entries deleted anyway by this code and changed by renamed old ones if any?

Reply:
Code comes from https://github.com/huggingface/datasets-server/pull/873. I think (from reviewing the preceding commits) that it was a way to fix a migration we had done wrong: job runners were already creating entries with the new name, and we applied the migration later, which lead to conflicts. Now, it should not occur, so, please feel free to remove this extra step!

Reply:
So sorry, I missed this thread. Sylvain added it on https://github.com/huggingface/datasets-server/pull/873, I think a similar case of duplicated key happened when renaming /split-names to /split-names-from-streaming.
But for some reason, it does not happened when renaming /first-rows to /split-first-rows-from-streaming